core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPreferWritePartitioning.java

```diff
@@ -93,6 +104,52 @@ public void testDoesNotFire()
                 .doesNotFire();
     }
 
+    @Test
+    public void testThreshold()
+    {
+        PlanNodeStatsEstimate stats = PlanNodeStatsEstimate.builder()
+                .addSymbolStatistics(ImmutableMap.of(new Symbol("col_one"), new SymbolStatsEstimate(0, 0, 0, 0, 1)))
+                .build();
+
+        assertPreferredPartitioning(Optional.of(new PartitioningScheme(
+                Partitioning.create(FIXED_HASH_DISTRIBUTION, ImmutableList.of(new Symbol("col_one"))),
+                ImmutableList.of(new Symbol("col_one")))))
+                .withSession(SESSION_WITH_PREFERRED_PARTITIONING_THRESHOLD)
+                .overrideStats("mock", stats)
+                .doesNotFire();
+    }
+
+    @Test
+    public void testThresholdWithHighNullFraction()
+    {
+        PlanNodeStatsEstimate stats = PlanNodeStatsEstimate.builder()
+                .addSymbolStatistics(ImmutableMap.of(new Symbol("col_one"), new SymbolStatsEstimate(0, 0, .99, 0, 100)))
+                .build();
+
+        assertPreferredPartitioning(Optional.of(new PartitioningScheme(
+                Partitioning.create(FIXED_HASH_DISTRIBUTION, ImmutableList.of(new Symbol("col_one"))),
+                ImmutableList.of(new Symbol("col_one")))))
+                .withSession(SESSION_WITH_PREFERRED_PARTITIONING_THRESHOLD)
+                .overrideStats("mock", stats)
+                .doesNotFire();
+    }
+
+    @Test
+    public void testThresholdWithMultiplePartitions()
+    {
+        PlanNodeStatsEstimate stats = PlanNodeStatsEstimate.builder()
+                .addSymbolStatistics(ImmutableMap.of(new Symbol("col_one"), new SymbolStatsEstimate(0, 0, 0, 0, 1)))
+                .addSymbolStatistics(ImmutableMap.of(new Symbol("col_two"), new SymbolStatsEstimate(0, 0, 0, 0, 2)))
+                .build();
+
+        assertPreferredPartitioning(Optional.of(new PartitioningScheme(
+                Partitioning.create(FIXED_HASH_DISTRIBUTION, ImmutableList.of(new Symbol("col_one"), new Symbol("col_two"))),
+                ImmutableList.of(new Symbol("col_one"), new Symbol("col_two")))))
+                .withSession(SESSION_WITH_PREFERRED_PARTITIONING_THRESHOLD)
+                .overrideStats("mock", stats)
+                .matches(MATCH_PATTERN);
```

Comment:
change name of `MATCH_PATTERN` to be more meaningful

Reply:
Changed to `SUCCESSFUL_MATCH`.